helix-lsp/src/client.rs

```diff
@@ -592,6 +597,25 @@ impl Client {
         )
     }
 
+    pub fn will_rename_files(
+        &self,
+        params: &Vec<lsp::FileRename>,
+    ) -> impl Future<Output = Result<lsp::WorkspaceEdit>> {
+        let files = params.to_owned();
```

Comment:
you need to check `workspace.fileOperations.willRename` herer (or even better at the callsite to support multiple LSPs)